packages/gitbeaker-core/src/services/Projects.ts

```diff
@@ -124,8 +124,10 @@ export class Projects extends BaseService {
     return RequestHelper.del(this, `projects/${pId}/fork`, options);
   }
 
-  search(projectName: string) {
-    return RequestHelper.get(this, 'projects', { search: projectName });
+  search(projectName: string, options?: BaseRequestOptions) {
+    return RequestHelper.get(this, 'projects', { search: projectName, ...options }) as Promise<
+      ProjectSchema[]
```

Comment:
I would add this return information to the function header instead of casting the return value. Besides that LGTM!!

Reply:
Ah I see ! I just did the same as in the all method but you're right :p

Reply:
Actually it is not that simple: 

```
search(projectName: string, options?: BaseRequestOptions): Promise<ProjectSchema[]>{
    return RequestHelper.get(this, 'projects', { search: projectName, ...options });
  }
```

raises the following error: `Type 'Promise<GetResponse>' is not assignable to type 'Promise<ProjectSchema[]>'`

Reply:
hmm, true. My bad, looks good then